api/chain.yml

```diff
@@ -0,0 +1,307 @@
+      tickets: 
+        type: array
+        items:
+          type: object
+          description: A Ticket is a marker of a tick of the blockchain's clock. It is the source of randomness for proofs of storage and leader election. It is generated by the miner of a block using a VRF and a VDF.
+          properties:
+            vrfProof:
+              type: string
+              description: A base64 encoded string representing the binary proof output of running a VRF on the VDFResult of the parent ticket.
+              example: QsIHRoYj...
+            vdfResult:
+              type: string
+              description: A base64 encoded string of data derived by running a VDF on vrfProof
+              example: cyBjbG9j...
+            vdfProof:
+              type: string
+              description: A base64 encoded string of the proof of delay during computation of vdfResult
+              example: nkgYSBwZ...
+      electionProof:
+        type: string
+        description: A base64 encoded byte array that represents the proof that the miner of this Block is a leader at this round.
+        example: V8896JC51ikqPVqYUCVbHzUfaOgjTBF6BYc8G/2P9CJMAg8MJzOi1j3F8hJSj/C6tKTLd9qHpLEjRpiGUzz04gE=
+      parentTipsetId:
+        type: string
+        description: The ID of the Tipset on which this block was based. In the spec this field is defined as an array of Block IDs, which is effectively equivalent to a Tipset ID in this API.
+        example: bafy2bzacebnsn4ywt2cgejgp32rgsv4vaylrqe32axw4cfadxhoutxxb7puzs
+      parentWeight:
+        type: number
+        description: parentWeight is the aggregate chain weight ([spec](https://github.com/filecoin-project/specs/blob/master/expected-consensus.md#chain-weighting)) of the parent Tipset.
+        example: 2001934
+      stateRoot:
+        type: string
+        description: The CID ([Content Identifier](https://github.com/ipld/cid)) of the root of the tree containing Actor state after the application of all the messages mined int his Block.
+        example: bafyreih4xzehg7qudgdwve27pvmtblnc3ib5ql7haeouainjnavyo46eoy
+      timestamp:
+        type: number
+        description: The block Timestamp is used to enforce a form of block delay by honest miners. Unix time UTC timestamp (in seconds) stored as an unsigned integer.
+        example: 1568753711
+      blsAggregate:
+        type: string
+        description: A base64 encoded representation of the aggregated BLS signature for all the messages in this block that were signed using BLS signatures
+        example: TODO
+      blockSig:
+        type: string
+        description: A base64 encoded representation of the signature over the hash of this entire block with the miner's worker key to ensure that it is not tampered with after creation
+        example: TODO
+  ExecutedMessage:
+    type: object
+    properties:
+      kind:
+        type: string
+        enum: [executedMessage]
+        readOnly: true
+      id:
```

Comment:
The executed message is an abstraction over a message + receipt for API convenience, and doesn't have an ID. The message itself does have a CID though.

I think the implication that you can't fetch an individual executed message is ok.